### PR TITLE
Allow `libffi` to work via `koch boot`

### PR DESCRIPTION
**Pull request by awr1**
cc @timotheecour + @yyoncho by way of #20168

Currently use of the CTFE VM FFI requires the libffi Nimble package as a compiler dep, which conflicts with the usage of the `--noNimblePath` option through `koch boot`. This can be worked around through just `cc`-ing `compiler.nim` directly (which is done in the tests), but this is a less convenient approach.

This solution is kinda messy: instead of using libffi as a dependency through Nimble we clone it like we do for tinyc + z3. I'm open to a better solution for this, the compiler itself has not been set up to mesh well with deps *through* Nimble (i.e. not via arbitrary clones) in spite of it technically being a Nimble package.

### CODE REVIEW

**Comment by Araq** [comment]
Will be merged after 2.0 has been released, too risky.

**Comment by awr1** [comment]
> Will be merged after 2.0 has been released, too risky.

Would that be for 2.2? I'll start a changelog

**Comment by awr1** [comment]
CI error is unrelated.

File: changelogs/changelog_2_2_0.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+## Standard library additions and changes
+
+## Language changes
+
+## Compiler changes
+
+## Tool changes
+
+- koch now allows bootstrapping with `-d:nimHasLibFFI`, replacing the older option of building the compiler directly w/ the `libffi` nimble package in tow.
+
```
Comment (ringabout) [inline changelogs/changelog_2_2_0.md:14]: It should be put in the top level changelog instead.

**Pull request merged**